pisa/stages/flux/pi_mceq_barr.py

```diff
@@ -499,9 +500,19 @@ def apply_sys_kernel(
         C = num gradients
     Not that first dimension (of length A) is vectorized out
     """
-    out[...] = (
-        nu_flux_nominal * spectral_index_scale(true_energy, energy_pivot, delta_index)
-    ) + np.dot(gradients, gradient_params)
+
+    # Nominal flux + spectral index change
+    result = nu_flux_nominal * spectral_index_scale(true_energy, energy_pivot, delta_index)
+
+    # Apply bar params
+    result += np.dot(gradients, gradient_params)
+
+    # Check for negative results from spline (np.clip not supported by vectorization)
+    negative_mask = result < 0
+    if np.sum(negative_mask):
```

Comment:
1. I'm guessing you mean to do `result[negative_mask] = 0.0` on L513?
2. Is the `if` clause on L512 premature / assumed optimization? Because it looks like the opposite to me. In the time you take to check every value in the mask (and you are even doing more than that; you're performing a _sum_ over the entire mask!), you might as well have just applied the mask.

So, unless you have actual timing tests run, I'm guessing it will be faster--and I know it removes LOC--if you simply do
```python
result[result < 0.0] = 0.0
```

Reply:
Fixed